### PR TITLE
exercise 3

### PR DESCRIPTION
**Pull request by michaelef123**
basic functionality is implemented. graphql-shield and -middleware are missing.
deleted unnecessary files in exercise 2 folder.

### CODE REVIEW

**Comment by roschaefer** [comment]
Remove `node_modules/` from your `master` branch.

Screenshot - 2019-12-02T020330 473

File: 3/test/excercise3.spec.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    const queryTodos = gql`query getTodos {todos { id, message, state }}`;
+
+    it('todo removed from list', () => {
+        mutate({mutation: mutateDeleteTodo, variables: {id: 3}}).then((result) => {
+            expect(result.data.deleteTodo).toBeTruthy();
+        })
+        query({query: queryTodos}).then((result) => {
+            expect(result.data.todos.length).toBe(6);
+        })
+    })
```
Comment (Xnasav) [inline 3/test/excercise3.spec.js:109]: Try to keep tests as capsulated/isolated as possible to reduce the risk of false-positives or false-negatives. Best would be to call a before_all function that is executed before all individual tests. 

grafik

File: 3/test/excercise3.spec.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    const querySignedAssignee = gql`query getSigneAssignee($token: String!) { assignees(token: $token) {id, name} }`;   
+    const tokenMessage = jwt.sign({auth: 'der richtige inhalt'}, 'shhhhh');
+
+    it('get all assignees signed', () => {
+        query({query: querySignedAssignee, variables: {token: tokenMessage}}).then((result) => {
+            expect(result.data.assignees.length).toBe(3);
+        })
```
Comment (Xnasav) [inline 3/test/excercise3.spec.js:30]: I am not sure about this one, but I think that JWTs belong into the request header and should not be provided in the body of the request

File: 3/resolvers.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          return null;
+        }
+      },
+      assignee: (parent, {id}) => { return assignees.filter(x => x.id = id); },
+      todos: () => { return todos; },
+      todo: (parent, {id} ) => { return todos.filter(x => x.id === id)[0]; },
+      todosForAssigneeId: (parent, {id}) => { return todos.filter(x => x.assigneeId === id); },
+      countTodos: () => { return todos != null? todos.length : 0; },
+      countAssignees: () => { return assignees != null? assignees.length : 0; },
+      todosByState: (parent, {state}) => { return todos.filter(x => x.state === state); }
```
Comment (Xnasav) [inline 3/resolvers.js:22]: I am not sure, but I think it is nor neccessary to implement individual queries for these functionalities. 

grafik
You could just use the GraphQL filter function

**Review by Xnasav — APPROVED** [review]

**Pull request merged**